lib/mobilePhoneNumberValidator.js

```diff
@@ -4,18 +4,38 @@
 // https://github.com/ruimarinho/google-libphonenumber/issues/92
 const phoneNumberUtil = require('google-libphonenumber').PhoneNumberUtil;
 const phoneNumberType = require('google-libphonenumber').PhoneNumberType;
-const phoneNumberFormat = require('google-libphonenumber').PhoneNumberFormat;
 
 function createValidatorService() {
+    function isCorrectShape(phoneNumber) {
+        return /^[0-9+][0-9 \-()+]+[0-9]$/.test(phoneNumber);
+    }
+
     function isValidMobilePhoneNumber(input) {
         try {
+            // decode unicode characters and then remove any, now unescaped, whitespace.
+            // this is inside the `try` block so that is deliberately throws when there is
+            // incomprehensible characters in the string. `decodeURIComponent` will fall
+            // over and throw an error.
+            // `\s` doesn't include the "zero-width space". explicitly removing it with
+            // the addition of `\u200B` in the regex.
+            const sanitisedInput = decodeURIComponent(input).replace(/[\s\u200B]/g, '');
+            if (!isCorrectShape(sanitisedInput)) {
+                return false;
+            }
             const phoneUtil = phoneNumberUtil.getInstance();
-            const number = phoneUtil.parse(input, 'GB');
-            phoneUtil.format(number, phoneNumberFormat.INTERNATIONAL);
-            if (phoneUtil.isValidNumber(number)) {
-                const inputtedPhoneNumberType = phoneUtil.getNumberType(number);
-                if (inputtedPhoneNumberType === phoneNumberType.MOBILE) {
-                    return true;
+            const number = phoneUtil.parseAndKeepRawInput(sanitisedInput, 'GB');
+            // formats it to UK mobile format with no leading `0`.
+            const formatted = phoneUtil.format(number);
+
+            // check if the end of the inputted string matches the formatted number.
+            // i.e. the user could enter "07...", or "+44...", or "7...". This will match
+            // all those scenarios.
+            if (input.replace(/[\s-]/g, '').endsWith(formatted.replace(/[\s-]/g, ''))) {
```

Comment:
Can this be removed? I reckon it's redundant and the tests are passing without it